app/src/main/java/mozilla/lockbox/store/RouteStore.kt

```diff
@@ -36,14 +45,40 @@ class RouteStore(
             .map(this::dataStoreToRouteActions)
             .filterNotNull()
             .subscribe(routes)
+
+        // do we want to onboard if they disconnect their account and
+        // then re-login?
+        dataStore.state
+            .subscribe { state ->
+                when (state) {
+                    is DataStore.State.Unprepared -> onboardingState.onNext(true)
+                    else -> onboardingState.onNext(false)
+                }
+            }
+            .addTo(compositeDisposable)
     }
 
     private fun dataStoreToRouteActions(storageState: DataStore.State): Optional<RouteAction> {
         return when (storageState) {
-            is DataStore.State.Unlocked -> RouteAction.ItemList
-            is DataStore.State.Locked -> RouteAction.LockScreen
-            is DataStore.State.Unprepared -> RouteAction.Welcome
-            else -> null
-        }.asOptional()
+            is DataStore.State.Unlocked -> chooseRoute()//RouteAction.ItemList
+            is DataStore.State.Locked -> return RouteAction.LockScreen.asOptional()
+            is DataStore.State.Unprepared -> return RouteAction.Welcome.asOptional()
+            // bad. figure out how to make this work again
+            else -> null!!
+        }
+    }
+
+    private var triggerOnboarding: Boolean? = null
+    private val triggerOnboardingSubject: Consumer<Boolean>
+        get() = Consumer { onboarding ->
+            triggerOnboarding = onboarding
+        }
```

Comment:
We should be avoiding having `Consumers` in stores. Stores should be listening to `dispatcher` for most of the time, and occasionally other stores.

AFAIUI, this should be a `Subject<Boolean>`.